rust/decentraland-godot-lib/src/godot_classes/dcl_ui_background.rs

```diff
@@ -0,0 +1,334 @@
+                    if let Some(slices) = self.current_value.texture_slices.as_ref() {
+                        (
+                            slices.left * texture_size.x,
+                            slices.top * texture_size.y,
+                            slices.right * texture_size.x,
+                            slices.bottom * texture_size.y,
+                        )
+                    } else {
+                        (
+                            texture_size.x / 3.0,
+                            texture_size.y / 3.0,
+                            texture_size.x / 3.0,
+                            texture_size.y / 3.0,
+                        )
+                    };
+
+                self.base.set_patch_margin(
+                    godot::engine::global::Side::SIDE_BOTTOM,
+                    patch_margin_bottom as i32,
+                );
+                self.base.set_patch_margin(
+                    godot::engine::global::Side::SIDE_LEFT,
+                    patch_margin_left as i32,
+                );
+                self.base.set_patch_margin(
+                    godot::engine::global::Side::SIDE_TOP,
+                    patch_margin_top as i32,
+                );
+                self.base.set_patch_margin(
+                    godot::engine::global::Side::SIDE_RIGHT,
+                    patch_margin_right as i32,
+                );
+
+                // TODO: should be AXIS_STRETCH_MODE_TILE or AXIS_STRETCH_MODE_STRETCH?
+                self.base.set_h_axis_stretch_mode(
+                    godot::engine::nine_patch_rect::AxisStretchMode::AXIS_STRETCH_MODE_TILE_FIT,
+                );
+                self.base.set_v_axis_stretch_mode(
+                    godot::engine::nine_patch_rect::AxisStretchMode::AXIS_STRETCH_MODE_TILE_FIT,
+                );
+            }
+            BackgroundTextureMode::Center => {
+                self.update_layout_for_center();
+            }
+            BackgroundTextureMode::Stretch => {
+                self.base
+                    .set_anchors_preset(godot::engine::control::LayoutPreset::PRESET_FULL_RECT);
+                self.base
+                    .set_patch_margin(godot::engine::global::Side::SIDE_BOTTOM, 0);
+                self.base
+                    .set_patch_margin(godot::engine::global::Side::SIDE_LEFT, 0);
+                self.base
+                    .set_patch_margin(godot::engine::global::Side::SIDE_TOP, 0);
+                self.base
+                    .set_patch_margin(godot::engine::global::Side::SIDE_RIGHT, 0);
+                self.base.set_h_axis_stretch_mode(
+                    godot::engine::nine_patch_rect::AxisStretchMode::AXIS_STRETCH_MODE_STRETCH,
+                );
+                self.base.set_v_axis_stretch_mode(
+                    godot::engine::nine_patch_rect::AxisStretchMode::AXIS_STRETCH_MODE_STRETCH,
+                );
+
+                // let image_size = godot_texture.get_size();
```

Comment:
Can we add a comment with the purpose of this commented code or delete if not more needed?